### PR TITLE
MM-12357: Add CLI command "team list"

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: sonasingh46 <sonasingh46@gmail.com>
#### Summary
The PR will add a cli command for listing commands under a team.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9490

### CODE REVIEW

File: cmd/mattermost/commands/command.go [hunk 1]
```diff
@@ -67,3 +76,33 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Enter at least one team")
+	}
+
+	teams := getTeamsFromTeamArgs(a, args)
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: I think would be nice to allow to list from all teams, like here: https://github.com/mattermost/mattermost-server/pull/9528/files#diff-a2d1b2bd4062c22fea5fdc1e88b50fe4R31

File: cmd/mattermost/commands/command.go [hunk 2]
```diff
@@ -67,3 +76,33 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Command().GetByTeam(team.Id); result.Err != nil {
+			CommandPrintErrorln("Unable to list commands for '" + args[i] + "'")
+		} else {
+			commands := result.Data.([]*model.Command)
+
+			for _, command := range commands {
+				CommandPrettyPrintln(command.DisplayName)
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: I think we need here the "command.Id" too, and probably, if we going to allow to list all slash commands from all teams, show the team name too. Something like fmt.Sprintf("%s: %s (team: %s)" , command.Id, command.DisplayName, team.Name)`.

Comment (sonasingh46) [reply cmd/mattermost/commands/command.go]: I will address this soon. @jespino 
Thanks for helping out.

File: cmd/mattermost/commands/command.go (hunk 1 above)
Comment (sonasingh46) [reply cmd/mattermost/commands/command.go]: @jespino 
I will address that.

File: cmd/mattermost/commands/command.go [hunk 3]
```diff
@@ -67,3 +76,33 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+		return errors.New("Enter at least one team")
+	}
+
+	teams := getTeamsFromTeamArgs(a, args)
+	for i, team := range teams {
+		if team == nil {
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Command().GetByTeam(team.Id); result.Err != nil {
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: We like to use idiomatic error handling as much as possible:

For example here, we would use something like:
```go
result := <-a.Srv.Store.Command().GetByTeam(team.Id)
if result.Err != nil {
    CommandPrintErrorln("Unable to list commands for '" + args[i] + "'")
    continue
}

commands := result.Data.([]*model.Command)
for _, command := range commands {
        CommandPrettyPrintln(command.DisplayName)
}
```

**Comment by sonasingh46** [comment]
@jespino 
I will update the PR soon.

**Comment by sonasingh46** [comment]
@jespino 
I have addressed the review comments.
PTAL

File: cmd/mattermost/commands/command.go [hunk 4]
```diff
@@ -67,3 +77,41 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+	if err != nil {
+
+		return err
+	}
+	defer a.Shutdown()
+
+	var teams []*model.Team
+	if len(args) < 1 {
+		teamList, err := a.GetAllTeams()
+		teams = teamList
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: It is more idiomatic to move this after the error handling.

File: cmd/mattermost/commands/command.go [hunk 5]
```diff
@@ -67,3 +77,41 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
...
+			continue
+		}
+		result := <-a.Srv.Store.Command().GetByTeam(team.Id)
+		if result.Err != nil {
+			CommandPrintErrorln("Unable to list commands for '" + args[i] + "'")
+			continue
+		}
+		commands := result.Data.([]*model.Command)
+		for _, command := range commands {
+			commadList := fmt.Sprintf("%s: %s (team: %s)", command.Id, command.DisplayName, team.Name)
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: Probably the name of the variable is confusing. maybe must be something like `commandListEntry` or `commandListItem`

File: cmd/mattermost/commands/command.go [hunk 6]
```diff
@@ -67,3 +77,41 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
 func moveCommand(a *app.App, team *model.Team, command *model.Command) *model.AppError {
 	return a.MoveCommand(team, command)
 }
+
+func listCommandCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+
```
Comment (jespino) [inline cmd/mattermost/commands/command.go]: Unnecesary extra empty line.

File: cmd/mattermost/commands/command.go (hunk 4 above)
Comment (sonasingh46) [reply cmd/mattermost/commands/command.go]: Correct.

**Comment by jespino** [comment]
@sonasingh46 this needs a rebase, please :)

**Comment by sonasingh46** [comment]
@jespino
I will do it asap.

**Comment by sonasingh46** [comment]
@jespino 
Rebase Done!

**Comment by jespino** [comment]
@sonasingh46 Merged! Thank you very much! Let me know if you want to help in anything else. We can talk about where do you want to help and find something specific for you :)

**Comment by sonasingh46** [comment]
@jespino 
Yes Sure! Currently I am exploring the project.
Whatever issue I can understand I am picking up so as to get started with the project.
After exploring it a bit i will ask you and can discuss some specific areas where I can contribute.

**Comment by jespino** [comment]
Awesome :)

**Pull request merged**